components/server/src/memory.rs

```diff
@@ -19,9 +19,24 @@ impl MemoryTraceManager {
             for id in ids {
                 let sub_trace = provider.sub_trace(id);
                 let sub_trace_name = sub_trace.name();
-                MEM_TRACE_SUM_GAUGE
-                    .with_label_values(&[&format!("{}-{}", provider_name, sub_trace_name)])
-                    .set(sub_trace.sum() as i64)
+                let leaf_ids = sub_trace.get_children_ids();
```

Comment:
How about
```rust
pub fn flush(&mut self, _now: Instant) {
        fn record = |trace: Arc<MemoryTrace>, names: &Vec<String>| {
            let name = trace.name();
            names.push(name);
            let subs = trace.get_children_ids();
            if subs.is_empty {
                MEM_TRACE_SUM_GAUGE
                    .with_label_values(&[&names.join("-")])
                    .set(trace.sum() as i64)
            } else {
                for sub in subs {
                    record(sub, names);
                }
            }
            names.pop();
        };
        let mut names = vec![];
        for provider in &self.providers {
            record(provider, &names);
        }
}
```

Reply:
In general, recursive function is not recommended in production code. We want to limit the max level of the metrics.